dali/test/python/test_dali_tf_dataset_pipelines.py

```diff
@@ -0,0 +1,271 @@
+
+    def __next__(self):
+        if self.n <= self.stop:
+            result = self.n
+            self.n += 1
+            ret = self.random_iter.next()[0]
+            return ret
+        else:
+            raise StopIteration
+
+
+class FixedSampleIterator:
+    def __init__(self, value):
+        self.value = value
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self.value
+
+class InfiniteSampleIterator:
+    def __init__(self, start_value):
+        self.value = start_value
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        result = self.value
+        self.value = self.value + np.array(1, dtype=self.value.dtype)
+        return result
+
+@pipeline_def
+def one_input_pipeline(def_for_dataset, device, source, external_source_device):
+    """Pipeline accepting single input via external source
+
+    Parameters
+    ----------
+    def_for_dataset : bool
+         True if this pipeline will be converted to TF Dataset
+    device : str
+        device that the Dataset will be placed ("cpu" or "gpu")
+    source : callable
+        callback for the external source in baseline pipeline otherwise None
+    external_source_device : str
+        Device that we want the external source in TF dataset to be placed
+    """
+    if def_for_dataset:
+        # We use no copy when the input memory is matching the external source placement,
+        # so the Dataset's placement is the same as external source's device
+        input = fn.external_source(name="input_placeholder",
+                                   no_copy=device == external_source_device,
+                                   device=external_source_device)
+    else:
+        input = fn.external_source(name="actual_input",
+                                   source=source,
+                                   batch=False,
+                                   device=external_source_device)
+    input = input if device == 'cpu' else input.gpu()
+    processed = fn.cast(input + 10, dtype=dali.types.INT32)
+    input_padded, processed_padded = fn.pad([input, processed])
+    return input_padded, processed_padded
+
+
+# Test that uses Tensor and Repeat (infinite) datasets as inputs to DALI pipeline
+def external_source_converter_with_fixed_value(shape, dtype, tensor):
+    def to_dataset(pipeline_desc, device_str):
+        with tf.device('/cpu:0'):
+            input_dataset = tf.data.Dataset.from_tensors(tensor).repeat()
+            # If we place DALIDataset on GPU we need the remote call + manual data transfer
+            if "gpu" in device_str:
+                input_dataset = input_dataset.apply(tf.data.experimental.copy_to_device('/gpu:0'))
+
+
+        dataset_pipeline, shapes, dtypes = pipeline_desc
+
+        with tf.device(device_str):
+            dali_dataset = dali_tf.experimental.DALIDatasetWithInputs(
+                    input_datasets=input_dataset,
+                    input_names="input_placeholder",
```

Comment:
I know it is something we don't normally do, but I think we could consider having some structure to pass this information, even a tuple would do `("input_placeholder", input_dataset, <maybe layouts and other data associated with this input>)`

Reply:
@mzient suggested a dictionary and a constructor:
```
{
  "images":dali.tf.input(ImageDataset, layout="HWC", batch=False),
  "labels":dali.tf.input(LabelDataset,  batch=False)
}
```
I will be working on that in a follow-up, I think it's the cleanest idea.